corehq/apps/hqwebapp/static/hqwebapp/js/crud_paginated_list.js

```diff
@@ -176,173 +230,147 @@ var CRUDPaginatedListModel = function (
-        if (data.rowClass) {
-            self.rowClass(data.rowClass);
+    ko.bindingHandlers.activeOnSimilar = {
+        update: function(element, valueAccessor, allBindingsAccessor) {
+            var current = valueAccessor()();
+            var compare = parseInt(allBindingsAccessor()['compareText']);
+            if (current === compare) {
+                $(element).addClass('active');
+            } else {
+                $(element).removeClass('active');
+            }
         }
     };
 
-    self.initTemplate = function (elems) {
-        var $updateForm = $(elems).find('.update-item-form');
-        if ($updateForm) {
-            $updateForm.submit(function (e) {
-                e.preventDefault();
-                $updateForm.ajaxSubmit({
-                    url: "",
-                    type: 'post',
-                    dataType: 'json',
-                    data: {
-                        action: 'update'
-                    },
-                    success: function (data) {
-                        if (data.updatedItem) {
-                            self.dismissModals();
-                            self.updateItemSpec(data.updatedItem);
-                        } else if (data.form) {
-                            var $updateForm = self.getItemRow().find('.update-item-form');
-                            if ($updateForm) {
-                                $updateForm.html($(data.form).html());
-                            }
-                        }
-                    }
-                });
-            });
-        }
-        var $deleteButton = $(elems).find('.delete-item-confirm');
-        if ($deleteButton) {
-            $deleteButton.click(function () {
-                $(this).button('loading');
-                self.getItemRow().trigger('deleteItem');
-            });
-        }
-        var $refreshButton = $(elems).find('.refresh-list-confirm');
-        if ($refreshButton) {
-            $refreshButton.click(function () {
-                $(this).button('loading');
-                self.getItemRow().trigger('refreshList');
-            });
-        }
-        self.initRow(elems, self);
+    $(function () {
+        var initial_page_data = hqImport("hqwebapp/js/initial_page_data").get,
+            paginatedListModel = hqImport("hqwebapp/js/crud_paginated_list").CRUDPaginatedListModel(
+                initial_page_data('total'),
+                initial_page_data('limit'),
+                initial_page_data('page'),
+                {
+                    statusCodeText: initial_page_data('status_codes'),
+                    allowItemCreation: initial_page_data('allow_item_creation'),
+                    createItemForm: initial_page_data('create_item_form'),
+                }
+            );
+
+        ko.applyBindings(paginatedListModel, $('#editable-paginated-list').get(0));
+        paginatedListModel.init();
+    });
+
+    return {
+        CRUDPaginatedListModel: CRUDPaginatedListModel,
```

Comment:
This file both returns module contents *and* sets up a document ready handler when it is first imported. Maybe I'm just not aware of how often this is done in our codebase, but this seems like exactly the scenario we were discussing trying to avoid in our discussion yesterday. I suppose it would be easy enough (although somewhat annoying) to split it into two modules: one library ("utility?") and one "main" module. Thoughts?

Reply:
Indeed, it's exactly that scenario. We do have a lot of modules that do this, but based on yesterday's conversation, I'm open to trying out splitting library and main modules and seeing how that feels.

Split it up in https://github.com/dimagi/commcare-hq/pull/19337/commits/cb4b8aac6bbf53bd9ca4a79e0856a3cd97bc0f9c  Went with `list` and `list_init` because the list module isn't exactly a utility, it's a model...maybe it should be `list_model` and `list_init`, but I don't really care right now, I imagine naming conventions will become more obvious as we try out this approach more and see what the common patterns are.